packages/wallet/WalletManager.js

```diff
@@ -0,0 +1,421 @@
+          this.account = new Account(this.network);
+          await this.account.setAccount(web3.eth.accounts.privateKeyToAccount(privateKey));
+          this.accountArray.push(Object.assign({}, this.account));
+          return this.account;
+        }
+        default:
+          console.log('The network is not set.');
+          return null;
+      }
+    }
+    if (arguments.length === 1 && Number.isInteger(arguments[0]) && arguments[0] >= 0) {
+      // generate the account of the wallet at the specified index
+      switch (this.network) {
+        case 'stellar': {
+          this.account = new Account(this.network);
+          await this.account.setAccount(this.wallet.getKeypair(arguments[0]));
+          this.accountArray.push(Object.assign({}, this.account));
+          return this.account;
+        }
+        case 'ethereum': {
+          const privateKey = `0x${this.wallet.deriveChild(arguments[0]).getWallet()._privKey.toString('hex')}`;
+          this.account = new Account(this.network);
+          await this.account.setAccount(web3.eth.accounts.privateKeyToAccount(privateKey));
+          this.accountArray.push(Object.assign({}, this.account));
+          return this.account;
+        }
+        default:
+          this.account = null;
+          console.log('The network has not been set.');
+      }
+      return this.account;
+    }
+    if (arguments.length === 1 && typeof (arguments[0]) === 'string') {
+      try {
+        switch (this.network) {
+          case 'stellar': {
+            if (arguments[0].charAt(0) === 'S') {
+              // generate account from private key
+              this.account = new Account(this.network);
+              await this.account.setAccount(StellarSdk.Keypair.fromSecret(arguments[0]));
+              this.accountArray.push(Object.assign({}, this.account));
+            } else {
+              this.account = null;
+              console.log('The starting char must be S (private key)');
+            }
+            break;
+          }
+          case 'ethereum': {
+            this.account = new Account(this.network);
+            await this.account.setAccount(web3.eth.accounts.privateKeyToAccount(arguments[0]));
+            this.accountArray.push(Object.assign({}, this.account));
+            break;
+          }
+          default:
+            this.account = null;
+            console.log('The network has not been set.');
+        }
+        return null;
+      } catch (err) {
+        console.log('The input is not a valid private/public key.');
+        return null;
+      }
+    } else {
+      console.log('The argument must be empty, or 0,1,2,...');
+      return null;
+    }
+  }
+
+  /**
+   * A shorthand for creating multiple accounts.
+   * The accounts created will be the 0th, 1st, ... (n-1)th in the wallet
+   * @param {int} numberOfAccounts
+   */
+  async setMultipleAccounts(numberOfAccounts) {
```

Comment:
Quite curious what this is for (no references found). Maybe it is more useful if it's a `getMultipleAccounts` instead?

Reply:
get an array of indexes -> accounts